docs/src/pages/components/text/TextPropControls.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+
+import {
+  Flex,
+  SelectField,
+  SliderField,
+  SwitchField,
+  TextField,
+  TextProps,
+  TextVariation,
+} from '@aws-amplify/ui-react';
+
+export interface TextPropControlsProps extends TextProps {
+  setAs: (value: React.SetStateAction<TextProps['as']>) => void;
+  setVariation: (value: React.SetStateAction<TextProps['variation']>) => void;
+  setIsTruncated: (
+    value: React.SetStateAction<TextProps['isTruncated']>
+  ) => void;
+  setColor: (value: React.SetStateAction<TextProps['color']>) => void;
+  setLineHeight: (value: React.SetStateAction<TextProps['lineHeight']>) => void;
+  setFontWeight: (value: React.SetStateAction<TextProps['fontWeight']>) => void;
+  setFontStyle: (value: React.SetStateAction<TextProps['fontStyle']>) => void;
+  setFontSize: (value: React.SetStateAction<TextProps['fontSize']>) => void;
+  setTextDecoration: (
+    value: React.SetStateAction<TextProps['textDecoration']>
+  ) => void;
+  value: string;
+  setValue: (value: string) => void;
+}
+
+interface TextPropControlsInterface {
+  (props: TextPropControlsProps): JSX.Element;
+}
+
+export const TextPropControls: TextPropControlsInterface = ({
+  variation,
+  setVariation,
+  as,
+  setAs,
+  isTruncated,
+  setIsTruncated,
+  color,
+  setColor,
+  lineHeight,
+  setLineHeight,
+  fontWeight,
+  setFontWeight,
+  fontSize,
+  setFontSize,
+  fontStyle,
+  setFontStyle,
+  textDecoration,
+  setTextDecoration,
+  value,
+  setValue,
+}) => {
+  const VARIATIONS_OPTIONS = [
+    'primary',
+    'secondary',
+    'tertiary',
+    'error',
+    'warning',
+    'info',
+    'success',
+  ];
+
+  const AS_OPTIONS = ['p', 'span', 'strong', 'em'];
+
+  return (
+    <Flex direction="column">
+      <TextField
+        label="Displayed Text"
+        onChange={(event) => setValue(event.target.value)}
+        defaultValue={value}
+      ></TextField>
+      <SelectField
+        name="variation"
+        value={String(variation)}
+        onChange={(event) => setVariation(event.target.value as TextVariation)}
+        label="variation"
+      >
+        {VARIATIONS_OPTIONS.map((option) => (
+          <option key={option} value={option}>
+            {option}
+          </option>
+        ))}
+      </SelectField>
+      <SelectField
+        name="as"
+        value={String(as)}
+        onChange={(event) => setAs(event.target.value as TextProps['as'])}
+        label="as"
+      >
+        {AS_OPTIONS.map((option) => (
```

Comment:
When I change the "as" field, I see that change reflected in the props to code, but not in the example TextField. If it does not visually affect the example TextField, is this prop worth including in the demo?

Reply:
Good catch. I thought about it and think the user may inspect it in devTool, so still useful